src/ui/sign_message_nbgl.c

```diff
@@ -0,0 +1,145 @@
+#ifdef HAVE_NBGL
+
+#include "io.h"
+#include "menu.h"
+#include "sol/parser.h"
+#include "sol/printer.h"
+#include "sol/print_config.h"
+#include "sol/message.h"
+#include "sol/transaction_summary.h"
+#include "glyphs.h"
+#include "apdu.h"
+#include "utils.h"
+
+#include "nbgl_page.h"
+#include "nbgl_use_case.h"
+
+#include "handle_sign_message.h"
+#include "handle_sign_offchain_message.h"
+
+// Layout of the review flow
+static nbgl_layoutTagValueList_t layout;
+// Used by NBGL to display the reference the pair number N
+static nbgl_layoutTagValue_t current_pair;
+
+// We will display at most 4 items on a Stax review screen
+#define MAX_SIMULTANEOUS_DISPLAYED_SLOTS 4
```

Comment:
Can't you display at most 5 ? :thinking:

Reply:
During review, there at most 9 lines, and therefore 4 pairs. https://github.com/LedgerHQ/ledger-secure-sdk/pull/222/files#diff-6a5e7e3d29e29f8b6ec66bd2b2b406356b1dc1e9f99c61ed924c90ac0d5a804fR43
I'm adding a PR to add this define in the SDK for the future.

Reply:
https://github.com/LedgerHQ/ledger-secure-sdk/pull/227
But not be accessible directly, will probably need to wait for next stax release.

Reply:
Yes it better be defined in the SDK it is error prone. I have defined this value as 5 in some of my apps. : l